### PR TITLE
Add experimental diagnostics ids

### LINKED ISSUE #4161: Define experimental diagnostics IDs
With https://github.com/dotnet/roslyn/pull/68702 and https://github.com/dotnet/runtime/pull/85444 we now have live experimental diagnostics (e.g., it's already surfaced in https://github.com/dotnet/aspnetcore/pull/49297), however, we don't have any diagnostic IDs defined. 

As per https://github.com/dotnet/extensions/pull/4130#issuecomment-1611882614, diagnostic IDs must be defined one per experimental feature. For example, generic math would use one ID that is applied to many APIs. The idea being that users can suppress usages on a per-feature level.

### PR DESCRIPTION
**Pull request by geeknoid**
Fixes #4161 

I also deleted a bunch of now useless analyzers (since they exist in the main Roslyn analyzers instead)
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4179)

### CODE REVIEW

**Review by rafal-mz — APPROVED** [review]

File: src/Shared/DiagnosticIds/Experiments.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable CA1716
+namespace Microsoft.Shared.DiagnosticIds;
+#pragma warning restore CA1716
+
+internal static class Experiments
```
[the remark this answers is not in the document]
Comment (geeknoid) [reply src/Shared/DiagnosticIds/Experiments.cs:14]: Done.

File: src/Shared/DiagnosticIds/Experiments.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+{
+#pragma warning disable S1075 // URIs should not be hardcoded
+    internal const string UrlFormat = "https://aka.ms/dotnet-extensions-warnings/{0}";
+#pragma warning restore S1075 // URIs should not be hardcoded
+
+    internal const string Resilience = "EXTEXP0001";
```
[the remark this answers is not in the document]
Comment (geeknoid) [reply src/Shared/DiagnosticIds/Experiments.cs:20]: Yeah, it's a bit uneven. But it's fine, given the use cases.

File: Directory.Build.targets [hunk 3]
```diff
@@ -35,9 +35,8 @@
     <!-- Prevent analyzer crashes from stopping things -->
```
Comment (RussKie) [inline Directory.Build.targets:35]: Lines 19-20 should have been deleted as well
image

File: src/Shared/DiagnosticIds/Experiments.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    internal const string Compliance = "EXTEXP0002";
+    internal const string Telemetry = "EXTEXP0003";
+    internal const string TimeProvider = "EXTEXP0004";
+    internal const string AutoClient = "EXTEXP0005";
+    internal const string AsyncState = "EXTEXP0006";
+    internal const string HealthChecks = "EXTEXP0007";
+    internal const string ResourceMonitoring = "EXTEXP0008";
+    internal const string Hosting = "EXTEXP0009";
+    internal const string ObjectPool = "EXTEXP0010";
+    internal const string DocumentDb = "EXTEXP0011";
```
Comment (RussKie) [inline src/Shared/DiagnosticIds/Experiments.cs:30]: We also need a doc listing all current IDs and explaining which API is each of these IDs light up, if enabled. 
The doc should also give guidance on what to do if the ID is encountered - for example, dotnet/aspnetcore has already encountered the TBD warning in the past (https://github.com/dotnet/aspnetcore/pull/49297#issuecomment-1629154246), so once these new IDs flow through what should the maintainers of the repo do. Is it safe to suppress, or should dotnet/aspnetcore rewrite something?

**Comment by RussKie** [comment]
@geeknoid @rafal-mz  we need to do the same for https://github.com/Azure/dotnet-extensions-experimental/ too.

**Pull request merged**